scope.go

```diff
@@ -482,6 +493,43 @@ func (s *scope) fullyQualifiedName(name string) string {
 	return fmt.Sprintf("%s%s%s", s.prefix, s.separator, name)
 }
 
+func (s *scope) uniqueID(name string, tags map[string]string) string {
+	name = s.fullyQualifiedName(name)
+	l := len(name)
+	for k, v := range tags {
+		l += len(k) + len(v)
+	}
+
+	// size the buffer to include separator bytes
+	buf := bytes.NewBuffer(make([]byte, 0, l+2*len(tags)+1))
+	buf.WriteString(name)
+
+	if len(tags) == 0 {
```

Comment:
We should move this check before the buffer creation so we don't need to create the buffer and return name immediately if there are no tags